libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -352,12 +352,10 @@ proc sendEncoded*(p: PubSubPeer, msg: seq[byte], isHighPriority: bool): Future[v
     f
   else:
     if len(p.rpcmessagequeue.nonPriorityQueue) >= p.maxNumElementsInNonPriorityQueue:
-      if not p.disconnected:
-        p.disconnected = true
-        libp2p_pubsub_disconnects_over_non_priority_queue_limit.inc()
-        p.closeSendConn(PubSubPeerEventKind.DisconnectionRequested)
-      else:
-        Future[void].completed()
+      p.behaviourPenalty += 0.0001
+      trace "Peer has reached maxNumElementsInNonPriorityQueue. Discarding message and applying behaviour penalty.", peer = p, score = p.score,
+        behaviourPenalty = p.behaviourPenalty, agent = p.getAgent()
+      Future[void].completed()
```

Comment:
Given most return paths are `Future[void].completed()` how about moving it outside the `if` statement and doing an explicit `return` on each arm for the cases that are not `Future[void].completed()`.

Reply:
There's nothing wrong per se. What I was hinting at was, when you have code akin to this:
```nim
if <cond>:
  foo()
  return A  

elif <cond>:
  bar()
  return A

elif <cond>:
  baz()
  return B

...

else:
  foobar()
  return A
```

Sometimes it's clearer to group/extract the returns:
```nim
if <cond>:
  foo()

elif <cond>:
  bar()


elif <cond>:
  baz()
  return B

...

else:
  foobar()

return A
```

But I just realised here it's almost 50/50, so not a lot of improvement can be made.